Oliver West

WestyCodes opened the pull request.

WestyCodes committed:
domain model, and spec to appease the great eslint

WestyCodes committed:
added test for create cohort
setup cohortmanager class, cohort class, and passed first test

WestyCodes committed:
added test for search cohort by name
wrote code to pass test
changed searchCohorts() to findCohort()

WestyCodes committed:
Wrote test for addStudentToCohort
needed a createStudent so i wrote a test and wrote code for that
cant commit with a failing test
 therefore finished code for addStudentToCohort

WestyCodes committed:
added test for remove cohort
wrote code to pass test

WestyCodes committed:
added tests for removeStuden
wrote code to pass tests

WestyCodes committed:
added test for findStudent
wrote code to pass test

WestyCodes committed:
added test for cohort capacity check
wrote code to pass test.

WestyCodes committed:
added tests for cohorts with no name or same name
wrote code to pass tests

WestyCodes committed:
Updated domain model
wrote test for findStudentByName
added code to pass test

WestyCodes committed:
implementing twilio sms

WestyCodes committed:
fixing env files

WestyCodes committed:
implemented twilio sms for add and remove

dogezen commented on spec/cohortManager.spec.js:
```diff
@@ -0,0 +1,218 @@
+  it('Should create a student with a unique ID', () => {
+    // setup
+    const student1 = new Student('John', 'Deere')
+    student1.studentId = 1
```
property for student could be named `id` so you write code like this `student1.id`

WestyCodes replied:
done!

dogezen commented on spec/cohortManager.spec.js:
```diff
@@ -0,0 +1,218 @@
+    // setup
+    cohortManager.createCohort('Cohort 1')
+    cohortManager.createCohort('Cohort 2')
+    const expected = `No cohort with this name`
```
refactor to throw new Error and catch these, if you feel like it

WestyCodes replied:
done! - all tests/code throw and catch errors.

dogezen commented on spec/cohortManager.spec.js:
```diff
@@ -0,0 +1,218 @@
+    cohortManager.createCohort('Cohort 1')
+    cohortManager.createStudent('John', 'Doe')
+    // execute
+    const result = cohortManager.addStudentToCohort(1, 'Cohort 1')
```
this line would be best written as `cohortManager.addStudentToCohort(student2.id, cohort1.name)` or similar

WestyCodes replied:
done!

dogezen commented on spec/cohortManager.spec.js:
```diff
@@ -0,0 +1,218 @@
+    // setup
+    const cohort1 = new Cohort('Cohort 1')
+    const expected = [cohort1]
+    cohortManager.createCohort('Cohort 1')
```
are your cohort names case sensitive? so can I create "Cohort 1" and "cohort 1"? or should I not be allowed to use hte same name?

WestyCodes replied:
all cohort names are forced into lower case now

dogezen commented on spec/cohortManager.spec.js:
```diff
@@ -0,0 +1,218 @@
+  })
+})
+
+describe('Cohort', () => {
```
move this into a separate tests file since youre testing cohort

WestyCodes replied:
done

dogezen commented on spec/cohortManager.spec.js:
```diff
@@ -0,0 +1,218 @@
+const { CohortManager, Cohort, Student } = require('./../src/cohortManager.js')
```
youd want some tests for Cohort and Student class, even just to make sure that the constructors save hte right data when an obj is created.

WestyCodes replied:
and also done!

WestyCodes committed:
changed studentId to id
updated addstudenttocohort test
changed tests and code so all cohort names are forced to lower case
moved cohort tests to cohort spec file

WestyCodes committed:
updated all code and tests to throw errors rather than return strings

WestyCodes committed:
added spec file for student class just to make sure it works as intended
